api/hasura/event-triggers/create-profile.ts

```diff
@@ -0,0 +1,54 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { gql } from '../../../api-lib/Gql';
+import { EventTriggerPayload } from '../../../api-lib/types';
+import { verifyHasuraRequestMiddleware } from '../../../api-lib/validate';
+import { profiles_constraint } from '../../../src/lib/gql/zeusHasuraAdmin';
+
+async function handler(req: VercelRequest, res: VercelResponse) {
```

Comment:
can you give some more details here @zashton ?

Reply:
the event trigger runs as a separate thread than the main mutation as its asynchronous, so the profile may get inserted only after the response is sent hence profile would be null for the response
@CryptoGraffe

Reply:
@zashton is this a cricitcal business requirement? The logic here is pretty simple regardless, based on my understanding. If the profile is null in the response, then it's a new profile and all that really exists in the database row is the eth address.

Reply:
> the event trigger runs as a separate thread than the main mutation as its asynchronous, so the profile may get inserted only after the response is sent hence profile would be null for the response

jumping in here after seeing this comment; apologies if i'm missing some context. but in general, it would be good to have a way to delay the response until we're sure that all triggers have completed, so that we can return data in the response that was created/modified by triggers.

Reply:
> jumping in here after seeing this comment; apologies if i'm missing some context. but in general, it would be good to have a way to delay the response until we're sure that all triggers have completed, so that we can return data in the response that was created/modified by triggers.
> 
> otherwise our client pattern gets more complicated: send request, wait for some time, fetch modified data, wait if it's not available yet, repeat...

@levity I think this is a question of how the new profile db entries are used, then. It doesn't seem like there is a situation where a profile needs to be synchronously accessible, since only a 3rd party creates a profile as a side effect of some other interaction. Therefore this seems like a great candidate for event-based business logic. Please correct me if this is not the case

If something does need a synchronous guarantee, then it should be included in the action logic directly.

Reply:
Also, is there any plan to migrate to graphQL subscriptions for state in the frontend?

Reply:
tried this mutation and indeed the profile was null, i'm not sure why `give_token_received` gets set correctly to 0 even though it is also an event trigger. 
```gql
mutation UpdateUserTest {
  adminUpdateUser(object: {address: "<address>", 
    new_address: "<new_address>", 
    circle_id: 1, non_receiver: true}) {
    id
    UserResponse {
      address
      pending_received_gifts{
      	tokens
        note
      }
      give_token_received
      profile {
        address
        id
      }
    }
  }
}

```

Reply:
> Also, is there any plan to migrate to graphQL subscriptions for state in the frontend?

this would be nice to have.